backends/dpdk/dpdkArch.cpp

```diff
@@ -1397,15 +1410,177 @@ const IR::Node* DismantleMuxExpressions::preorder(IR::Mux* expression) {
     return path2;
 }
 
-cstring DismantleMuxExpressions::createTemporary(const IR::Type* type) {
+const IR::Node* DismantleMuxExpressions::postorder(IR::AssignmentStatement* statement) {
+    if (statements.empty())
+        return statement;
+    statements.push_back(statement);
+    auto block = new IR::BlockStatement(statements);
+    statements.clear();
+    return block;
+}
+
+bool SplitHSIndexExpression::hasHSE(const IR::Expression *hse) {
+    if (auto mem = hse->to<IR::Member>()) {
+        if (auto ai = mem->expr->to<IR::ArrayIndex>()) {
+            if (!ai->right->is<IR::Constant>())
+                return true;
+        }
+    }
+    return false;
+}
+
+// This function drives the replacement of statements with variable array index expressions
+// with a series of if statements based on the number of elements in the header stack.
+void SplitHSIndexExpression::replaceVarIndexWithIf(IR::AssignmentStatement *statement) {
```

Comment:
Did you check whether we don't already have a midend pass for this? I recall reviewing one.

Reply:
Oh! I do see a pass in midend,  _HSIndexTransform_
It is not used in dpdk backend. I will try invoking that and see it can be used here.
Thanks for the pointer!!